### PR TITLE
system: replace confirm() use for consistency

### PR DESCRIPTION
**Pull request by fichtner**

### CODE REVIEW

File: src/opnsense/www/js/opnsense_widget_manager.js [hunk 1]
```diff
@@ -564,18 +565,38 @@ class WidgetManager  {
         return $panel;
     }
 
-    _restoreDefaults(confirmDialog = true) {
-        if (confirmDialog) {
-            if (!confirm(this.gettext.restoreconfirm)) {
-                return;
-            }
-        }
+    __restoreDefaults(dialog) {
```
Comment (fichtner) [inline src/opnsense/www/js/opnsense_widget_manager.js:568]: only C programmers appreciate this ;)

File: src/opnsense/www/js/opnsense_widget_manager.js [hunk 2]
```diff
@@ -322,6 +322,7 @@ class WidgetManager  {
                 message: $content,
                 buttons: [{
                     label: this.gettext.add,
+                    cssClass: 'btn-primary',
```
Comment (fichtner) [inline src/opnsense/www/js/opnsense_widget_manager.js:325]: in theory this is nicer but primary color for header and button doesn't match which is annoying. we could also pivot to "warning" dialogs here? not sure if "yellow" is better.. at least better than "red" (danger)

Comment (fichtner) [reply src/opnsense/www/js/opnsense_widget_manager.js]: I removed the primary hint in the modals so this is now just a consistency fix for the `confirm()` replacement.

**Review by swhite2 — APPROVED** [review]

File: src/opnsense/www/js/opnsense_widget_manager.js [hunk 3]
```diff
@@ -564,18 +565,38 @@ class WidgetManager  {
...
+        BootstrapDialog.show({
+            title: this.gettext.restore,
+            draggable: true,
+            animate: false,
+            message: this.gettext.restoreconfirm,
+            buttons: [{
+                label: this.gettext.ok,
+                cssClass: 'btn-primary',
+                hotkey: 13,
+                action: async (dialog) => {
```
[the remark this answers is not in the document]
Comment (fichtner) [reply src/opnsense/www/js/opnsense_widget_manager.js]: done, works fine

**Pull request merged**